### ISSUE TITLE
Extra quotes added when replacing with multi-cursor

### ISSUE THREAD

**Issue by nicholasmarais1158** [zed-industries/zed#14349]
### Description

To replace the double quotes in this example with backticks;

1. Start with a cursor after one of them and then add a second after the other
2. Backspace the double quotes
3. Add backticks

Notice that an extra backtick is added after the second. This is not the case in VSCode.

https://github.com/user-attachments/assets/4568aa3b-2907-4341-8d5d-a40a4cf5cfea

### Environment

Zed: v0.143.7 (Zed)
OS: macOS 14.5.0
Memory: 36 GiB
Architecture: aarch64

```
const fav = "Zed"
const q = "Your favorite? ${fav}"
```

**Comment by notpeter** [comment]
Yeah, this seems like a bad interaction with multi-cursor and autosurround/autoclose.

<img width="715" alt="image" src="https://github.com/user-attachments/assets/232ebfda-4731-4a41-b6e5-3afa7fce741d">

**Comment by benjaminwelch** [comment]
This is happening with quotes (perhaps other characters as well) even with a single cursor:

![zed-quote-insert-bug](https://github.com/user-attachments/assets/8f4ec8a7-75c3-4495-a455-43e1c843bddc)

It affects both single/double quotes for me if I type them out manually. It should autoclose the first one I type but it doesn't, then closes the second one.
